src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -79,6 +79,10 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 return String.Empty;
 
             Contract.Assert(payloadName.Count == payload.Count);
+            if(payloadName.Count != payload.Count)
```

Comment:
It is still valid.  However, we don't want the process to crash, so this is to make sure that things don't fail.